### PR TITLE
Add the DKTC dataset (Extension of #3564)

### PR DESCRIPTION
**Pull request by sooftware**
New pull request of #3564. (for DKTC)

### CODE REVIEW

File: datasets/dktc/README.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+Here is an example from the dataset:
+```
+{'class': 0,
+'text': '이 약이 뭔 거 같아? \n뭔데요.?\n뭐긴 뭐야 네 엄마 저승길 보내주는 약이지.\n네? 그게 무슨.\n네 엄마 지금 저 빌딩에 있지? 그리고 저 빌딩에선 시음회가 열리고 있고.\n그게.무슨.\n이미 인력은 다 배치해놨어. 저 음료를 마시면 네 엄마는.\n아 안 돼!!\n네 엄마 살리고 싶으면 내가 시키는 대로 해.\n멈춰요 알겠으니까 그만하라고!!\n내가 시키는 대로 해!!\n알겠다고요!!! 제발 그만'}
+```
+
+### Data Fields
+
+{The dataset has two fields:
```
Comment (lhoestq) [inline datasets/dktc/README.md:80]:
```suggestion
The dataset has two fields:
```

File: datasets/dktc/README.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+```bibtex
+@misc{DKTC
+  author       = {Cho, Soyoung and Ha, Sangchun and Ryu, Myeonghyeon and
+                  Keum, Bitna and Park, Kyubyong},
+  title        = {DKTC, Dataset of Korean Threatening Conversations},
+  howpublished = {https://github.com/tunib-ai/DKTC},
+  year         = {2022},
+}
+```
```
Comment (lhoestq) [inline datasets/dktc/README.md:156]:
````suggestion
```

### Contributions

Thanks to [@sooftware](https://github.com/sooftware) for adding this dataset.
````

**Review by lhoestq — COMMENTED** [review]
Cool thanks ! 

Can you run `make style` to fix the code formatting and also regenerate the `dataset_infos.json` to account for the new ClassLabel type ?

Also it looks like the `dummy_data.zip` file is empty. Can you try to recreate it please ? It must contain a dummy version of `train.csv`, with only a few lines from the original csv file.

**Comment by sooftware** [comment]
I reflect your comment! @lhoestq

**Comment by sooftware** [comment]
I update the code according to the dataset structure change.

**Comment by lhoestq** [comment]
Thanks ! I think the dummy data are not valid yet - the dummy train.csv file only contains a partial example (the quotes `"` start but never end).

File: datasets/dktc/dktc.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  author       = {Cho, Soyoung and Ha, Sangchun and Ryu, Myeonghyeon and
+                  Keum, Bitna and Park, Kyubyong},
+  title        = {DKTC, Dataset of Korean Threatening Conversations},
+  howpublished = {https://github.com/tunib-ai/DKTC},
+  year         = {2022},
+}
+"""
+
+_DESCRIPTION = """
+It is a Korean dataset that classifies threatening conversation data into four threat situations.
```
Comment (lhoestq) [inline datasets/dktc/dktc.py:33]: remove trailing whitespace
```suggestion
It is a Korean dataset that classifies threatening conversation data into four threat situations.
```

File: datasets/dktc/dktc.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+}
+"""
+
+_DESCRIPTION = """
+It is a Korean dataset that classifies threatening conversation data into four threat situations. 
+
+- 0: threat
+- 1: extortion
+- 2: workspace
+- 3: etc
```
Comment (lhoestq) [inline datasets/dktc/dktc.py:38]: remove trailing whitespace
```suggestion
- 3: etc
```

**Comment by sooftware** [comment]
> Thanks ! I think the dummy data are not valid yet - the dummy train.csv file only contains a partial example (the quotes `"` start but never end).

Hi! @lhoestq There is a problem.  

  
As shown in the picture above, the conversation is divided into "\n" in the "conversion" column.  
That's why there's an error in the file path that only saved only five lines like below. 

```
'idx', 'class', 'conversation'
'0', '협박 대화', '"지금 너 스스로를 죽여달라고 애원하는 것인가?'
아닙니다. 죄송합니다.'
죽을 거면 혼자 죽지 우리까지 사건에 휘말리게 해? 진짜 죽여버리고 싶게.'
정말 잘못했습니다.
```
   
In fact, these five lines are all one line.

**Comment by sooftware** [comment]
Sorry, I'm late to check! I'll send it to you soon!

**Comment by albertvillanova** [comment]
Thanks for your contribution, @sooftware. Are you still interested in adding this dataset?

We are removing the dataset scripts from this GitHub repo and moving them to the Hugging Face Hub: https://huggingface.co/datasets

We would suggest you create this dataset there, under this organization namespace: https://huggingface.co/tunib

Please, feel free to tell us if you need some help.

**Comment by sooftware** [comment]
Close this PR. Thanks!

**Pull request closed without merging**